src/auxiliary/math.jl

```diff
@@ -5,6 +5,65 @@
 @muladd begin
 #! format: noindent
 
+"""
+    sqrt(x::Real)
+
+Custom square root function which returns `NaN` for negative arguments instead of throwing an error.
+This is required to ensure [correct results for multithreaded computations](https://github.com/trixi-framework/Trixi.jl/issues/1766) 
+when using the [`Polyester` package](https://github.com/JuliaSIMD/Polyester.jl), 
+i.e., using the `@batch` macro instead of the Julia built-in `@threads` macro, see [`@threaded`](@ref).
```

Comment:
Do you also want to link the Discourse thread (here and in the other docstring) or is not valuable enough?

Reply:
ok /r